File: src/main/java/net/dean/jraw/models/attr/Votable.java
package net.dean.jraw.models.attr;

import net.dean.jraw.models.Thing;
import net.dean.jraw.models.VoteDirection;
import net.dean.jraw.models.meta.JsonProperty;

/**
 * Indicates that this model can be voted on
 *
 * @see net.dean.jraw.managers.AccountManager#vote(Thing, VoteDirection)
 */
public interface Votable extends JsonAttribute {
    /**
     * The net score of this Votable model
     *
     * @return The link's net score
     */
    @JsonProperty
    Integer getScore();


    /**
     * Gets the way in which the logged in user voted. If there is none,
     * {@link net.dean.jraw.models.VoteDirection#NO_VOTE} will be returned.
     *
     * @return The way in which the logged in user voted
     */
    @JsonProperty
    VoteDirection getVote();
}

File: lib/src/main/kotlin/net/dean/jraw/models/Votable.kt
package net.dean.jraw.models

/**
 * Common interface for models that can be upvoted or downvoted
 */
interface Votable {
    /** How the logged-in user voted on the model */
    val vote: VoteDirection

    /** Upvotes minus downvotes */
    val score: Int

    /** Upvotes & downvotes */
    val ups: Int
    val downs: Int
}
